File: Assets/Game/Scripts/Moduls/PS_FourFlames.lua
---@class PS_FourFlames : ModulBase 
PS_FourFlames = ModulBase:extend()

function PS_FourFlames:new()
  self.Flame_A = nil ---@type GameObject
  self.Flame_B = nil ---@type GameObject
  self.Flame_C = nil ---@type GameObject
  self.Flame_D = nil ---@type GameObject
end
function PS_FourFlames:Start()
  self:Deactive()
end
function PS_FourFlames:Active()
  self.Flame_A:SetActive(true)
  self.Flame_B:SetActive(true)
  self.Flame_C:SetActive(true)
  self.Flame_D:SetActive(true)
end
function PS_FourFlames:Deactive()
  self.Flame_A:SetActive(false)
  self.Flame_B:SetActive(false)
  self.Flame_C:SetActive(false)
  self.Flame_D:SetActive(false)
end
function PS_FourFlames:Reset()
end
function PS_FourFlames:Backup()
end

function CreateClass_PS_FourFlames()
  return PS_FourFlames()
end

File: Assets/Game/GamePlay/Scripts/Moduls/PS_FourFlames.cs
using UnityEngine;

namespace Ballance2.Game.GamePlay.Moduls
{
  /// <summary>
  /// 起点四火焰组件
  /// </summary>
  public class PS_FourFlames : ModulBase 
  {
    public GameObject Flame_A;
    public GameObject Flame_B;
    public GameObject Flame_C;
    public GameObject Flame_D;

    public override void Active()
    {
      base.Active();
      this.Flame_A.SetActive(true);
      this.Flame_B.SetActive(true);
      this.Flame_C.SetActive(true);
      this.Flame_D.SetActive(true);
    }
    public override void Deactive()
    {
      base.Deactive();
      this.Flame_A.SetActive(false);
      this.Flame_B.SetActive(false);
      this.Flame_C.SetActive(false);
      this.Flame_D.SetActive(false);
    }

    public override void ActiveForPreview()
    {
      this.Flame_A.SetActive(true);
      this.Flame_B.SetActive(true);
      this.Flame_C.SetActive(true);
      this.Flame_D.SetActive(true);
    }
    public override void DeactiveForPreview()
    {
      this.Flame_A.SetActive(false);
      this.Flame_B.SetActive(false);
      this.Flame_C.SetActive(false);
      this.Flame_D.SetActive(false);
    }
  }
}